src/metabase-report-notifier/src/main/java/com/champtitles/metabasereportexecutor/notifier/EmailSender.java

```diff
@@ -0,0 +1,216 @@
+     * @param fileName   name of the xlsx file attachment
+     * @param xlsxData   contents of the xlsx file attachment
+     */
+    public void sendEmail(String subject, String[] recipients, String fileName, byte[] xlsxData) {
+        logger.info("creating email message");
+        Message message = new MimeMessage(session);
+        setFromAddress(message);
+        setRecipients(message, recipients);
+        setSubject(message, subject);
+        setAttachment(message, fileName, xlsxData);
+        send(message);
+    }
+
+    /**
+     * Sends the email message using the transport
+     *
+     * @param message email message to send
+     */
+    private void send(Message message) {
+        try {
+            logger.info("sending email");
+            emailTransport.send(message);
+
+        } catch (MessagingException e) {
+            logger.error("failed to send email message");
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Set the FROM field in the email message
+     *
+     * @param message email message
+     */
+    private void setFromAddress(Message message) {
+        try {
+            logger.info("using from address: {}", fromAddress);
+            message.setFrom(new InternetAddress(fromAddress));
+
+        } catch (MessagingException e) {
+            logger.error("invalid from address: {}", fromAddress);
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Set the list of recipients in the email message
+     *
+     * @param message    email message
+     * @param recipients list of recipient email addresses
+     */
+    private static void setRecipients(Message message, String[] recipients) {
+        for (String recipient : recipients) {
+            try {
+                logger.info("adding recipient: {}", recipient);
+                message.addRecipient(Message.RecipientType.TO, new InternetAddress(recipient));
+
+            } catch (MessagingException e) {
+                logger.error("invalid recipient address: {}", recipient);
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    /**
+     * Set the subject line in the email message
+     *
+     * @param message email message
+     * @param subject subject line to use
+     */
+    private static void setSubject(Message message, String subject) {
+        try {
+            logger.info("setting email subject: {}", subject);
+            message.setSubject(subject);
+
+        } catch (MessagingException e) {
+            logger.error("invalid subject text: {}", subject);
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Add the xlsx file attachment to the email message
+     *
+     * @param message  email message
+     * @param fileName name of the xlsx file
+     * @param xlsxData data contents of the xlsx file
+     */
+    private static void setAttachment(Message message, String fileName, byte[] xlsxData) {
```

Comment:
This looks like it creates the body and add the attachments to it, rather than just adding attachments?

Reply:
Ill update this method